package/gluon-core/luasrc/lib/gluon/upgrade/200-wireless

```diff
@@ -214,13 +214,14 @@ wireless.foreach_radio(uci, function(radio, index, config)
 		uci:set_list('wireless', radio_name, 'hostapd_options', hostapd_options)
 
 		if is_outdoor() then
-			uci:set('wireless', radio_name, 'channels', config.outdoor_chanlist())
+			if not wireless.preserve_channels(uci) then
+				uci:set('wireless', radio_name, 'channels', config.outdoor_chanlist())
+			end
 
 			-- enforce outdoor channels by filtering the regdom for outdoor channels
 			uci:set('wireless', radio_name, 'country3', '0x4f')
 			configure_mesh_wireless(radio, index, config, true)
 		else
-			uci:delete('wireless', radio_name, 'channels')
```

Comment:
This still needs to be wrapped in a condition to only apply in case the channel should not be preserved. Otherwise, without channel preservation the configuration could become invalid.

Reply:
In which case?
```
channel '1'
channels '2-3'
```
is a valid configuration and will choose channel 1. This could lead to confusion if outdoor mode is turned on, then off and the `channels` configuration still exists, but I think it's even more confusing if the `channels` option is removed for 5G devices when it's not removed for 2G devices.
Another possibility would be to remove the `channels` option for 2G devices, too (without `preserve_channels`).

Reply:
I think I have to wrap my head around that a little more.

The issue is, we only support `channels` in outdoor-mode, as the mesh-interface is only configured using netifd scripts which only utilize the `channel` setting.

Reply:
On the other hand - one can also configure channel 99999 and choose to preserve it's settings, which will also result in broken configuration.

Given we don't expose this key in the UI - I think we should just prevent all altering of channel settings from upgrade-scripts in case this option is enabled.

Reply:
So should I leave this as-is or should I add the if-condition?

Reply:
Looking at it, i think we need a different approach

1. Refactoring the code flow, moving the the "channel" / "channels" configuration to a different dedicated function
2. Only calling this function in case preserve-channels is not activated.